plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -167,6 +190,9 @@ public IcebergSplitSource(
                     continue;
                 }
             }
+            if (!partitionMatchesConstraint(identityPartitionColumns, partitionValues.get(), constraint)) {
```

Comment:
The Supplier's `get` is always getting called here, which kinda defeats the purpose. You can add an `constraint.predicate().isPresent()` check first, before calling `partitionMatchesConstraint`, or pass the entire Supplier to  `partitionMatchesConstraint` rather than calling `get` here to prolong constructing the Map until actually necessary.